File: java/src/main/java/it/giannibombelli/wsc2026/payment/domain/payment/PaymentId.java
package it.giannibombelli.wsc2026.payment.domain.payment;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.domain.identity.AggregateId;

import java.util.UUID;

public record PaymentId(UUID value) implements AggregateId {
    public PaymentId {
        Require.requireArgument(value, "PaymentId value");
    }
}

File: typescript/src/payment/domain/payment/paymentId.ts
// Identità dell'aggregato Payment.

import { requireArgument } from '@/common/utils/requireArgument.js';
import type { AggregateId } from '@/common/domain/identity/aggregateId.js';
import { Uuid } from '@/common/domain/primitive/uuid.js';

export class PaymentId implements AggregateId {
  readonly value: Uuid;

  constructor(value: Uuid) {
    requireArgument(value, 'PaymentId value');
    this.value = value;
  }
}
